File: Data Structures & Algorithms/special-array-with-x-elements-greater-than-or-equal-x/submission-3.py
class Solution:
    def specialArray(self, nums):

        nums.sort()
        n = len(nums)

        for i in range(n):
            x = n - i

            if nums[i] >= x and (i == 0 or nums[i - 1] < x):
                return x

        return -1

File: Data Structures & Algorithms/sort-array-by-parity/submission-3.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {

        int left = 0;
        int right = nums.size() - 1;

        while (left < right) {

            // Find an odd number on the left
            while (left < right && nums[left] % 2 == 0) {
                left++;
            }

            // Find an even number on the right
            while (left < right && nums[right] % 2 == 1) {
                right--;
            }

            // Swap odd on left with even on right
            if (left < right) {
                swap(nums[left], nums[right]);
                left++;
                right--;
            }
        }

        return nums;
    }
};